repositories/convoy_repository.go

```diff
@@ -0,0 +1,48 @@
+package repositories
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/checkmarble/marble-backend/models"
+	convoy "github.com/frain-dev/convoy-go/v2"
+)
+
+type convoyResources interface {
+	GetClient() *convoy.Client
+}
```

Comment:
For now, instead of consumming directly `*convoy.Client` (as Firebase, Metabase do) I inject a "getter" interface.

IMO, it will help define a dynamic "getter" if required in the future (eg, separate client per application (transfercheck, marble core...) or no client when no license). Still, this is too abstract to fully decide if it's a good choice or not.